File: core/src/bara/game/components/BehaviorComponent.java
package bara.game.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.ashley.core.Entity;
import com.badlogic.gdx.ai.btree.BehaviorTree;
import com.badlogic.gdx.ai.btree.decorator.AlwaysFail;
import com.badlogic.gdx.ai.btree.decorator.AlwaysSucceed;
import com.badlogic.gdx.utils.Pool;
import bara.game.behavior.Behavior;

public class BehaviorComponent implements Component, Pool.Poolable {
    public Behavior behavior;

    @Override
    public void reset() {
        behavior = null;
    }

}

File: core/src/bara/game/components/BehaviorComponent.kt
package bara.game.components

import bara.game.ashley.GameComponent
import bara.game.behavior.Behavior
import bara.game.ashley.EngineEntity

class BehaviorComponent : GameComponent() {
    var behavior: Behavior? = null

    override fun reset() {
        behavior = null
    }

    override fun buildOnto(engineEntity: EngineEntity) {
        val tempBehavior = behavior //temp to get around name shadowing
        engineEntity.with<BehaviorComponent> {
            behavior = tempBehavior
        }
    }

}
